### ISSUE TITLE
Action cloning and ChawatheScriptGenerator performance

### ISSUE THREAD

**Issue by EikeStein** [GumTreeDiff/gumtree#171]
I am executing a pathing algorithm with each node in the path-graph represented by an ITree. In order to find new nodes and manipulating them I frequently need to clone the trees (easily done via ITree.deepCopy()). For the expansion of the graph I need to analyze the actions from the current node to the destination node. However, the performance of the ChawatheScriptGenerator is not high enough to generate the actions of each deepCopy again. But if I don't do that the ITrees of the action still reference the original and not the cloned tree which makes finding those nodes/trees very difficult and error prone. 
My current solution is to try and "reassign" the actions to the cloned tree, but so far the results are far from bugfree. 

To summarize:

1. A lot of clones are created for each node in the graph
2. For each clone I need the actions to the destination tree 
3. If the action match a certain pattern I create a new node in my search-graph and want to execute the actions on a new clone or regenerate the actions from the new clone to the destination (that takes too long)

Any help is greatly appreciated!

**Comment by jrfaller** [comment]
I am not sure to understand what you want ;)

You think there is a performance problem with ChawatheScriptGenerator?

**Comment by jrfaller** [comment]
Hi!

Any follow-up on this issue?

**Comment by EikeStein** [comment]
Sorry for not responding. The issue has been resolved, I just misunderstood the API at the time. Thanks!
